File: C/idades.c
#include <stdio.h>
#include <locale.h>

char nome1[100], nome2[100];
int idade1, idade2;
double media;

int main() {
    setlocale(LC_ALL, "Portuguese");

    printf("Dados da primeira pessoa: \n");
    printf("Nome: ");
    gets(nome1);
    printf("Idade: ");
    scanf("%d", &idade1);

    printf("Dados da segunda pessoa: \n");
    printf("Nome: ");
    fseek(stdin, 0, SEEK_END);
    gets(nome2);
    printf("Idade: ");
    scanf("%d", &idade2);

    media = (double)(idade1 + idade2) / 2;
    printf("A idade média de %s e %s é de %.1lf anos", nome1, nome2, media);
}

File: Java/idades.java
import java.util.Locale;
import java.util.Scanner;

public class idades {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        Locale.setDefault(Locale.US);

        System.out.println("Dados da primeira pessoa: ");
        System.out.print("Nome: ");
        String nome1 = sc.nextLine();
        System.out.print("");
        System.out.print("Idade: ");
        int idade1 = sc.nextInt();

        sc.nextLine();

        System.out.printf("Dados da segunda pessoa: %n");
        System.out.print("Nome: ");
        String nome2 = sc.nextLine();
        System.out.print("");
        System.out.print("Idade: ");
        int idade2 = sc.nextInt();

        double media = (double)(idade1 + idade2) / 2;

        System.out.printf("A idade média de %s e %s é de %.1f anos", nome1, nome2, media);

        sc.close();

    }
}
